src/sage/quadratic_forms/binary_qf.py

```diff
@@ -1596,17 +1596,20 @@ def small_prime_value(self, Bmax=1000):
                 raise ValueError("Unable to find a prime value of %s" % self)
             B += 10
 
-    def solve_integer(self, n, *, algorithm="general"):
+    def solve_integer(self, n, *, algorithm="general", _flag=2):
         r"""
         Solve `Q(x, y) = n` in integers `x` and `y` where `Q` is this
         quadratic form.
 
         INPUT:
 
-        - ``n`` -- a positive integer
+        - ``n`` -- a positive integer or a
+          `:sage:`~sage.structure.factorization.Factorization` object
 
         - ``algorithm`` -- ``"general"`` (default) or ``"cornacchia"``
 
+        - ``_flag`` -- ``1``, ``2`` (default) or ``3``
```

Comment:
The underscore suggests this already, but perhaps you could say explicitly that this argument is for internal use only? This way we could skip a deprecation when #37364 eventually gets done.